scripts/run_metrics.sh

```diff
@@ -38,43 +65,68 @@ if [ ! -f "$FOCAL_OS_RAW_IMAGE" ]; then
 fi
 
 pushd $WORKLOADS_DIR
-grep focal sha1sums-x86_64 | sha1sum --check
+grep focal sha1sums-${TEST_ARCH} | sha1sum --check
 if [ $? -ne 0 ]; then
     echo "sha1sum validation of images failed, remove invalid images to fix the issue."
     exit 1
 fi
 popd
 
+if [ ${TEST_ARCH} == "aarch64" ]; then
+    build_fio
+
+    # Update the fio in the cloud image to use io_uring on AArch64
+    FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_NAME="focal-server-cloudimg-arm64-custom-20210929-0-update-tool.raw"
+    cp "$FOCAL_OS_RAW_IMAGE" "$WORKLOADS_DIR/$FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_NAME"
+    FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_ROOT_DIR="$WORKLOADS_DIR/focal-server-cloudimg-root"
+    if [ ! -d "$FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_ROOT_DIR" ]; then
+        mkdir -p "$FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_ROOT_DIR"
+    fi
+    # Mount the 'raw' image, replace the fio and umount the working folder
+    guestmount -a "$WORKLOADS_DIR/$FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_NAME" -m /dev/sda1 "$FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_ROOT_DIR" || exit 1
+    cp "$WORKLOADS_DIR"/fio "$FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_ROOT_DIR"/usr/bin/fio
+    guestunmount "$FOCAL_OS_RAW_IMAGE_UPDATE_TOOL_ROOT_DIR"
+fi
+
 # Build custom kernel based on virtio-pmem and virtio-fs upstream patches
-VMLINUX_IMAGE="$WORKLOADS_DIR/vmlinux"
+if [ ${TEST_ARCH} == "aarch64" ]; then
+       VMLINUX_IMAGE="$WORKLOADS_DIR/Image"
+else
+       VMLINUX_IMAGE="$WORKLOADS_DIR/vmlinux"
+fi
 
 LINUX_CUSTOM_DIR="$WORKLOADS_DIR/linux-custom"
 
 if [ ! -f "$VMLINUX_IMAGE" ]; then
     SRCDIR=$PWD
     pushd $WORKLOADS_DIR
     time git clone --depth 1 "https://github.com/cloud-hypervisor/linux.git" -b "ch-5.15.12" $LINUX_CUSTOM_DIR
-    cp $SRCDIR/resources/linux-config-x86_64 $LINUX_CUSTOM_DIR/.config
+    cp $SRCDIR/resources/linux-config-${TEST_ARCH} $LINUX_CUSTOM_DIR/.config
     popd
 fi
 
 if [ ! -f "$VMLINUX_IMAGE" ]; then
```

Comment:
You guys know that this won't rebuild the kernel if the config or tree has changed?

Reply:
yeah I thought this is intentional... Hi @likebreath, is this correct?

Reply:
Currently the peformance-metrics tests for x86-64 run on a VM which means there is a new disk every time.

Reply:
Hi @rbradford, personally I think if this is a problem that needs to be fixed, the fix should belong to another PR, as this fix is out of scope of this PR.

Can we finish this PR first and then I will send another PR improving the `run_metrics` scripts? Thanks!

Reply:
> Hi @rbradford, personally I think if this is a problem that needs to be fixed, the fix should belong to another PR, as this fix is out of scope of this PR.
> 
> Can we finish this PR first and then I will send another PR improving the `run_metrics` scripts? Thanks!

I agree with that! Thanks.